Sources/PostgreSQL/Connection/PostgreSQLConnection+NotifyAndListen.swift

```diff
@@ -0,0 +1,29 @@
+import Async
+
+extension PostgreSQLConnection {
+    public func listen(
+        _ channelName: String,
+        handler: @escaping (String) throws -> ()
+        ) throws -> Future<Void> {
+        beforeClose = { conn in
+            let query = PostgreSQLQuery(query: "UNLISTEN \(channelName);")
+            return conn.send([.query(query)], onResponse: { _ in })
+        }
+        let query = PostgreSQLQuery(query: "LISTEN \(channelName);")
+        return queue.enqueue([.query(query)], onInput: { message in
+            switch message {
+            case let .notificationResponse(notification):
+                try handler(notification.message)
```

Comment:
Related to the above: Sorry for the stupid question, but is it guaranteed that all payloads sent to that particular channel (but not those sent to other channels) are handled by this onInput handler? I guess yes, because the handler returns `false`, but then see the above comment.

Explanation: I have no idea whether PostgreSQLConnection is „smart“ enough to automatically dispatch all subsequent messages to this channel to this onInput handler, or whether the handler would be discarded after having received its first notification.

Reply:
https://www.postgresql.org/docs/10/static/sql-listen.html it's in the manual